backend/src/main/java/fk/prof/backend/model/assignment/AggregationWindowPlanner.java

```diff
@@ -0,0 +1,203 @@
+package fk.prof.backend.model.assignment;
+
+import com.google.common.base.Preconditions;
+import fk.prof.aggregation.model.FinalizedAggregationWindow;
+import fk.prof.backend.aggregator.AggregationWindow;
+import fk.prof.backend.model.aggregation.AggregationWindowLookupStore;
+import fk.prof.backend.proto.BackendDTO;
+import fk.prof.backend.util.BitOperationUtil;
+import fk.prof.backend.util.proto.RecorderProtoUtil;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import recording.Recorder;
+
+import java.time.Clock;
+import java.time.LocalDateTime;
+import java.util.Random;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class AggregationWindowPlanner {
```

Comment:
Package movement once existing forks for e2e testing are merged and v1 has released. This is to avoid unnecessary merge conflicts